roles/maintenance_32_apache/tasks/main.yml

```diff
@@ -0,0 +1,278 @@
+  when: "'HTTPD_ROOT' in apache_details.stdout"
+  changed_when: false
+
+# Extract main server config file path
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract server config file path"
+  ansible.builtin.set_fact:
+    apache_config_file_name: "{{ apache_root_directory + '/' + apache_details.stdout | regex_search('SERVER_CONFIG_FILE=\"([^\"]+)\"') | replace('\"', '') | split('=') | last }}"
+  changed_when: false
+
+# Extract whether SSL is enabled
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract whether SSL is enabled"
+  ansible.builtin.shell: "{{ apache_binary_path }} -M | grep ssl"
+  register: apache_mod_ssl_output
+  ignore_errors: true
+  changed_when: false
+
+# Extract whether SSL is enabled
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract whether SSL is enabled"
+  ansible.builtin.set_fact:
+    apache_is_ssl_active: "{{ 'ssl_module' in apache_mod_ssl_output.stdout }}"
+  changed_when: false
+
+# Extract all config file regexes
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract config path regexes from main apache config"
+  ansible.builtin.shell:
+    cmd: "grep -r '^Include' {{ apache_config_file_name }} | awk '(substr($2,0,1) != \"/\"){printf \"{{ apache_root_directory + '/' }}\"}{print $2}'"
+  register: apache_included_config_paths_out
+  ignore_errors: true
+  changed_when: false
+
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract config path regexes from main apache config"
+  ansible.builtin.set_fact:
+    apache_included_config_paths: "{{ apache_included_config_paths_out.stdout_lines }}"
+  ignore_errors: true
+  changed_when: false
+
+# Find all included error log files in all configs
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract log paths from apache configs"
+  ansible.builtin.shell:
+    cmd: "grep -rhE '^ErrorLog\\|^CustomLog' {{ item }} | awk '{print $2}'"
+  loop: "{{ apache_included_config_paths }}"
+  register: apache_log_files_out
+  changed_when: false
+  ignore_errors: true
+
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract log paths from apache configs"
+  ansible.builtin.set_fact:
+    apache_log_files: "{{ apache_log_files_out.results | map(attribute='stdout_lines') | flatten }}"
+  changed_when: false
+  ignore_errors: true
+
+# Add the default log paths to the list
+# There might also be module specific ones in this list
+- <<: *task
+  vars:
+    taskid: 32-009
+    name: "Apache: Extract log paths from apache configs"
+  ansible.builtin.set_fact:
+    apache_log_files: "{{ apache_log_files | default([]) + [
+                  '/var/log/{{ apache_binary_name }}/error_log',
+                  '/var/log/{{ apache_binary_name }}/access_log',
+                  '/var/log/{{ apache_binary_name }}/ssl_engine_log',
+                  '/var/log/{{ apache_binary_name }}/rewrite.log',
+                  '/var/log/{{ apache_binary_name }}/debug.log']
+                }}"
```

Comment:
Nesting Jinja templates (`{{ }}` within `{{ }}`) are not interpolated - string concatenation would need to be used instead. Here it'd literally say e.g. `/var/log/{{ apache_binary_name }}/error_log` in the resulting list.

Besides, where are these "default" logfile names from? At least on Debian, they are definitely not any kind of default. I'm also not quite clear on the reasoning behind this static list - shouldn't these be present in the config files anyway?

Unless there's a reason against it, I'd propose to remove this "defaults list" entirely.

Reply:
List is removed. But log file gathering needed "a few" extensions to get something useful out of it on debian and el based systems.